paasta_tools/paasta_cli/cmds/rollback.py

```diff
@@ -33,44 +35,70 @@ def add_subparser(subparsers):
         help='Rollback a docker image to a previous deploy')
 
     list_parser.add_argument('-k', '--commit',
-                             help='Git sha to mark for rollback',
+                             help='Git SHA to mark for rollback',
                              required=True,
                              )
-    list_parser.add_argument('-i', '--instance',
-                             help='Mark the instance we want to roll back (e.g. '
-                             'canary, .main)',
-                             required=True,
+    list_parser.add_argument('-i', '--instances',
+                             help='Mark one or more instances to roll back (e.g. '
+                             '"canary", "canary,main"). If no instances specified,'
+                             ' all instances for that service are rolled back',
+                             default='',
+                             required=False,
                              ).completer = lazy_choices_completer(list_instances)
     list_parser.add_argument('-c', '--cluster',
-                             help='Mark the cluster we want to rollback (e.g. '
-                             'cluster1, cluster2)',
+                             help='Mark the cluster to rollback (e.g. '
+                             'cluster1)',
                              required=True,
                              ).completer = lazy_choices_completer(list_clusters)
     list_parser.add_argument('-s', '--service',
-                             help='Name of service you wish to rollback'
+                             help='Name of the service to rollback (e.g. '
+                             'service1)'
                              ).completer = lazy_choices_completer(list_services)
 
     list_parser.set_defaults(command=paasta_rollback)
 
 
+def validate_given_instances(service_instances, args_instances):
+    """Trying to find which instances are actually valid for the given service"""
```

Comment:
Try and be more specific in your docstrings. Here's an example:

Given two lists of instances, return the intersection and difference of those two lists.

:param service_instances: the list of instances belonging to a services
:param args_instances: the desired instances
:returns: a tuple of (common, difference), indicating instances common to the two lists, and those only in args_instances